gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/map/Concat.feature

```diff
@@ -0,0 +1,213 @@
+# specific language governing permissions and limitations
+# under the License.
+
+@StepClassMap @StepConcat
+Feature: Step - concat()
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXa_bX_concat
+    Given the empty graph
+    And the traversal of
+      """
+      g.inject("a", "b").concat()
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | a |
+      | b |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXa_bX_concat_XcX
+    Given the empty graph
+    And the traversal of
+      """
+      g.inject("a", "b").concat("c")
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | ac |
+      | bc |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXa_bX_concat_Xc_dX
+    Given the empty graph
+    And the traversal of
+      """
+      g.inject("a", "b").concat("c", "d")
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | acd |
+      | bcd |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXa_bX_concat_Xinject_c_dX
+    Given the empty graph
+    And the traversal of
+      """
+      g.inject("a", "b").concat(__.inject("c", "d"))
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | acda |
+      | bcdb |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXListXa_bXcX_concat_XdX
+    Given the empty graph
+    And using the parameter xx1 defined as "l[a,b]"
+    And the traversal of
+      """
+      g.inject(xx1,"c").concat("d")
+      """
+    When iterated to list
+    Then the traversal will raise an error with message containing text of "String concat() can only take string as argument"
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXaX_concat_Xinject_List_b_cX
+    Given the empty graph
+    And using the parameter xx1 defined as "l[b,c]"
+    And the traversal of
+      """
+      g.inject("a").concat(__.inject(xx1))
+      """
+    When iterated to list
+    Then the traversal will raise an error
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXnullX_concat_XinjectX
+    Given the empty graph
+    And the traversal of
+      """
+      g.inject(null).concat()
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | null |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXnull_aX_concat_Xnull_bX
+    Given the empty graph
+    And the traversal of
+      """
+      g.inject(null, "a").concat(null, "b")
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | b |
+      | ab |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXhello_hiX_concat_XV_valuesXnameXX
```

Comment:
as i mentioned elsewhere, this one feels off to me. i wouldn't expect the traversal given to `concat()` to be fully iterated so:

```
| result |
| hellomarko |
| himarko |
```

Reply:
>  The reasoning for getting it fully iterated is treating the traversal results like the string inputs.

i think my main problem with this is that i can't think of any steps that work that way. no steps allow you to expand a traversal's results as multiple arguments. i'm wary of introducing that, even if it sorta makes sense and is convenient because then you have to know which step does an expansion and which just takes the first value as a single argument.

Reply:
> i think my main problem with this is that i can't think of any steps that work that way. no steps allow you to expand a traversal's results as multiple arguments. i'm wary of introducing that, even if it sorta makes sense and is convenient because then you have to know which step does an expansion and which just takes the first value as a single argument.
Ah yes, that is true as well. I will revert to taking the first value.